__test__/git-auth-helper.test.ts

```diff
@@ -437,15 +439,16 @@ describe('git-auth-helper tests', () => {
     }
   )
 
-  const configureSubmoduleAuth_configuresTokenWhenPersistCredentialsTrueAndSshKeyNotSet =
-    'configureSubmoduleAuth configures token when persist credentials true and SSH key not set'
+  const configureSubmoduleAuth_configuresSubmodulesWhenPersistCredentialsFalseAndSshKeyNotSet =
```

Comment:
some of the tests for configureSubmoduleAuth were redundant so i collapsed based on input combinations:
- `persist-credentials: false`, `ssh-key: ''`
- `persist-credentials: false`, `ssh-key: 'set'`
- `persist-credentials: true`, `ssh-key: ''`
- `persist-credentials: true`, `ssh-key: 'set'`

Reply:
Each test just tests the submoduleForeach calls so it made sense to collapse.